src/passes/GlobalTypeOptimization.cpp

```diff
@@ -444,7 +444,7 @@ struct GlobalTypeOptimization : public Pass {
 
             // Clear the old names and write the new ones.
             nameInfo.fieldNames.clear();
-            for (Index i = 0; i < oldFieldNames.size(); i++) {
+            for (Index i = 0; i < indexesAfterRemoval.size(); i++) {
               auto newIndex = indexesAfterRemoval[i];
               if (newIndex != RemovedField && oldFieldNames.count(i)) {
```

Comment:
Could use `oldFieldNames.find()` to remove the number of map lookups while we're here.